### PR TITLE
Convars

### PR DESCRIPTION
**Pull request by alt4**
I figured ConVars would be much nicer to set the mod's options, letting users set them through their autoexec without diving into the scripts and allowing hosts to have different configurations on different instances without deploying different versions of the mod.

I'm not sure the implementation I did is optimal however, since it differs slightly from the Bob's method which involves using [structs](https://github.com/R2Northstar/NorthstarMods/blob/7652539f68a35bd78e27e4394cf66a269212febb/Northstar.CustomServers/mod/scripts/vscripts/lobby/_private_lobby.gnut#L5) to contain [em](https://github.com/R2Northstar/NorthstarMods/blob/7652539f68a35bd78e27e4394cf66a269212febb/Northstar.CustomServers/mod/scripts/vscripts/lobby/_private_lobby.gnut#L15), not quite sure why.

Anyway I added every vars you documented in the README to mod.json, and updated the appropriate inits. The only thing that didn't make the cut is adding rules, because I'm not sure how to deal with that (pretty sure it's impossible to create `pv_rule_x` convars where `x` is the rule number sadly).

Marked as WIP for now because I haven't got the opportunity to playtest it yet!

### CODE REVIEW

**Comment by ScureX** [comment]
Thank you for your contribution, as you can [see](https://github.com/ScureX/Takyon.PlayerVote/projects/1#card-76495701) i have wanted some persistence but never got around to actually doing it, so this is a great start! This would be great for v3.0.0 (or v2.3.0, unsure if im not a bit fast with versioning lmk what you think).

---

**Few things:**
- In ```takyon_votekick.nut``` and ```takyon_voteextend-nut``` under ```CheckIfEnoughExtendVotes()```/ ```CheckIfEnoughSkipVotes()``` I'm currently using half/quarter as options. Its sucks and there should actually a var for that, is it possible for you to handle that with one convar for each file?
---
- In ```ReadMe.md``` the default value for ```pv_admin_names``` is wrong and differs from the actual code
- In ```ReadMe.md``` I'd like something along the lines of "values dont persist on full server restart, to change default values go to mod.json change this and that"
---
- In ```mod.json``` the defaultvalue for ```pv_admin_names``` contains a placeholder. Showing how to add admins should be done in a comment instead! This value might also change once i move to UIDs instead of names so dont put too much work into it just yet :D
- In ```mod.json``` some comments are incorrect (mostly bools saying true and false instead of 0 and 1), would be good to fix this before release
---
- Playtesting is not a problem we can do that together or i can do it with someone else, as long as you've checked that it doesn't crash at everything 
---
Again, thank you for contributing and don't stress the fixes too much! For help etc. hmu on discord :)

**Comment by alt4** [comment]
> unsure if im not a bit fast with versioning

Well if you're planning to follow semver religiously they [do recommend](https://semver.org/#if-even-the-tiniest-backwards-incompatible-changes-to-the-public-api-require-a-major-version-bump-wont-i-end-up-at-version-4200-very-rapidly) doing major versions increments each and every time a breaking change occurs, even if that makes you spit out hundreds of versions.

Adding convars likely wouldn't break stuff outright but would force hosts to move their settings from the scripts to their autoexec, so I'd go for 3.0.0.

> half/quarter as options

So basically adding a convar (well, one for extend and another for kick) to decide the percentage of players that need to vote before it is successful? If so, yeah sure!

I'm slowly adding the mod to my own servers; doing well so far!

**Comment by ScureX** [comment]
>  I'd go for 3.0.0

Then I'll do v3.0.0. I'd like all of this in a v3.0.0 branch instead of main (till release), don't know how much sense that makes and if that's even possible for you tho.


> to decide the percentage of players that need to vote before it is successful

Exactly (But for extend and skip, kick already has that implemented)! That'd require some reworking of the if-statement, but it shouldn't be too complicated, you can see how i did it in ```takyon_votekick.nut```

Glad it's doing well, genuinely hope there are no crashes ^^'

**Comment by ScureX** [comment]
I resolved the conflicts which arose because i removed the condition to only execute code on server, see [here](https://github.com/ScureX/Takyon.PlayerVote/projects/1#card-76878570). Should work fine now!

**Comment by alt4** [comment]
Should be about everything! Running it on a listenning server with no issues, and also deploying it to my public servers for a trial by fire.

**Review by ScureX — APPROVED** [review]
